android/src/main/java/com/pspdfkit/flutter/pspdfkit/PspdfkitPlugin.java

```diff
@@ -90,4 +160,32 @@ private boolean isImageDocument(@NonNull String documentPath) {
         }
         return extension.equals("png") || extension.equals("jpg") || extension.equals("jpeg");
     }
+
+    @Override
+    public boolean onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        int status = 0;
+        String permission = permissions[0];
+        if (requestCode == 0 && grantResults.length > 0) {
+            if (ActivityCompat.shouldShowRequestPermissionRationale(registrar.activity(), permission)) {
+                // Permission denied.
+                status = 1;
+            } else {
+                if (ActivityCompat.checkSelfPermission(registrar.context(), permission) == PackageManager.PERMISSION_GRANTED) {
+                    // Permission allowed.
+                    status = 2;
+                } else {
+                    // Set to never ask again.
+                    Log.i(LOG_TAG, "Set to never ask again " + permission);
+                    status = 3;
+                }
+            }
+        }
+        Log.i(LOG_TAG, "Requesting permission status: " + status);
+        Result result = this.result;
```

Comment:
Why did you not write it like this:

```
if (result != null) {
  result.success(status);
}
result = null;
```

Is there an advantage to setting `this.result` to `null` before calling `success`?

Reply:
It was an attempt to mitigate the issue of sending twice `result.success(status)`. Calling `success` twice on the same result is bad and would throw an exception.

Unfortunately it didn't work well after performing more stress tests:
<img width="1426" alt="Screen Shot 2019-08-01 at 11 01 29 AM" src="https://user-images.githubusercontent.com/3308503/62283485-f0fd6600-b451-11e9-8d02-c8b72dc9c0ab.png">

**How to observe the issue:**
Run the example in the `README` file, and deny the permission.
Quickly press `Tap to open document` until it crashes.

![tap-to-open-document](https://user-images.githubusercontent.com/3308503/62283891-d5468f80-b452-11e9-935a-eabcd81c1219.gif)

**What's going on under the hood:**
When the `requestPermission` method is quickly called many times the risk is to have the `onRequestPermissionsResult` callback that sends the result `result.success(status);` that has been already sent because the `permissionRequestResult` was not yet nullified.

I'm going to improve this implementation using a more solid atomic variable 🤘